var/spack/repos/builtin/packages/uqtk/package.py

```diff
@@ -14,32 +14,59 @@ class Uqtk(CMakePackage):
     url = "https://github.com/sandialabs/UQTk/archive/v3.0.4.tar.gz"
     git = "https://github.com/sandialabs/UQTk.git"
 
+    maintainers("omsai", "bjdebus")
+
     version("master", branch="master")
+    version("3.1.3", sha256="37840630357c4f407191d7a4276dfe219df35d54d288d68fea1746dfcbc3c5c1")
+    version("3.1.2", sha256="57ce0cea709777cbefb46f3bd86a0996a0ed5f50fc54cc297599df6e4bb9ab83")
     version("3.1.0", sha256="56ecd3d13bdd908d568e9560dc52cc0f66d7bdcdbe64ab2dd0147a7cf1734f97")
     version("3.0.4", sha256="0a72856438134bb571fd328d1d30ce3d0d7aead32eda9b7fb6e436a27d546d2e")
 
     variant(
-        "pyuqtk", default=True, description="Compile Python scripts and interface to C++ libraries"
+        "python", default=True, description="Compile Python scripts and interface to C++ libraries"
     )
 
     depends_on("expat")
-    depends_on("sundials", when="@3.1.0:")
+    depends_on("sundials@6:", when="@3.1.3:")
+    depends_on("sundials@:5", when="@3.1.0:3.1.2")
     depends_on("blas", when="@3.1.0:")
     depends_on("lapack", when="@3.1.0:")
 
-    extends("python", when="+pyuqtk")
-    depends_on("py-numpy", when="+pyuqtk")
-    depends_on("py-scipy", when="+pyuqtk")
-    depends_on("py-matplotlib", when="+pyuqtk")
-    depends_on("py-pymc3", when="+pyuqtk")
-    depends_on("swig", when="+pyuqtk")
+    extends("python", when="+python")
+    depends_on("py-numpy", when="+python")
+    depends_on("py-scipy", when="+python")
+    depends_on("py-matplotlib", when="+python")
+    depends_on("py-pymc3", when="+python")
+    depends_on("swig", when="+python")
```

Comment:
Are these dependencies documented anywhere? Most of these should be `type=("build", "run")`, although swig is usually `type="build"`

Reply:
@adamjstewart thanks for catching the `swig` build time only dependency issue.  Indeed, there isn't a `requirements.txt` file for the uqtk python extension whereby one might quickly verify the python extension dependency list and any minimum python dependency versions. My understanding is that some of those dependencies (for example `py-pymc3`) are meant to support the examples and are not part of the core package, so maybe they should only be pulled in with a `+examples` variant (I checked that several spack packages also use an "examples" variant) when enabling the `+python` variant. I'll comb through the package and make those changes.

> Or you can comment on the build.

@tldahlgren yes, my mistake! I appreciate your nudge for reviewers and I'm sorry for getting ahead of myself!

Reply:
@adamjstewart -

1. I've fixed swig to be a build-time only dependency.  Also the swig dependency was removed after @3.1.0 per sandialabs/UQTk@370e8a77744dc290aafdb115a1f066e8cf5698c5 (those directories contained the swig wrapper files)
2. I had a video conference with @bjdebus yesterday and the python wrapper is a work in progress. Eventually it will become its own package on pypi. Currently, as you saw, there isn't any `setup.py` or `requirements.txt`, etc in the python wrapper to check dependencies against, but I did confirm that the current python dependencies are all required and limited to those packages by checking the imports and excluding the example programs:
   ```console
   $ git clone https://github.com/sandialabs/UQTk
   $ cd UQTk/
   $ find PyUQTk/ \
          -name '*.py' \
          -not -path 'PyUQTk/pytests/*' \
          -exec grep -h -w import {} + |
          grep -E -v '[[:space:]]*#|[[:space:]]*from [.]' |
          awk '{$1=$1;print}' |
          sort | uniq
   ```
   ... and also running all the Python unit tests which I've included in the latest commit message of 0ff1316 (The 9 python tests among the 31 total tests start with Py*)

So everything looks okay to me.

Reply:
Python deps still need to be `type=("build", "run")` at least

Reply:
@adamjstewart Whoops, my mistake!  Sorry for not reading your first comment more carefully.  I've fixed the python deps with your instruction.  Also, I squashed the commits to reduce the noise in the log.